hypertrace-gradle-docker-java-application-plugin/src/main/resources/application-start-script.template.sh

```diff
@@ -23,6 +23,4 @@
 
 set -e
 
-DEFAULT_JVM_OPTS=${defaultJvmOpts}
```

Comment:
we could have used `DEFAULT_JVM_OPTS = (${defaultJvmOpts.substring(1, defaultJvmOpts.length()-1)})" if it was bash. Since sh doesn't support arrays, need to directly template it in exec command

Reply:
Currently before this change all the jvm args are transformed into one string `'"arg1" "arg2"` via gradle template transformer

Reply:
so are we trimming this to become `"arg1" "arg2"` then (the outer quotes only)? I'm curious how this works in the default template since it certainly looks quite hacky.

Reply:
Considerably more complex, worth taking a look - just want to make sure we're not deviating ourselves into brittle code:

```
DEFAULT_JVM_OPTS=${defaultJvmOpts}

# Collect all arguments for the java command:
#   * DEFAULT_JVM_OPTS, JAVA_OPTS, JAVA_OPTS, and optsEnvironmentVar are not allowed to contain shell fragments,
#     and any embedded shellness will be escaped.
#   * For example: A user cannot expect \${Hostname} to be expanded, as it is an environment variable and will be
#     treated as '\${Hostname}' itself on the command line.

set -- \\
<% if ( appNameSystemProperty ) {
     %>        "-D${appNameSystemProperty}=\$APP_BASE_NAME" \\
<% } %>        -classpath "\$CLASSPATH" \\
<% if ( mainClassName.startsWith('--module ') ) {
     %>        --module-path "\$MODULE_PATH" \\
<% } %>        ${mainClassName} \\
        "\$@"

# Stop when "xargs" is not available.
if ! command -v xargs >/dev/null 2>&1
then
    die "xargs is not available"
fi

# Use "xargs" to parse quoted args.
#
# With -n1 it outputs one arg per line, with the quotes and backslashes removed.
#
# In Bash we could simply go:
#
#   readarray ARGS < <( xargs -n1 <<<"\$var" ) &&
#   set -- "\${ARGS[@]}" "\$@"
#
# but POSIX shell has neither arrays nor command substitution, so instead we
# post-process each arg (as a line of input to sed) to backslash-escape any
# character that might be a shell metacharacter, then use eval to reverse
# that process (while maintaining the separation between arguments), and wrap
# the whole thing up as a single "set" statement.
#
# This will of course break if any of these variables contains a newline or
# an unmatched quote.
#

eval "set -- \$(
        printf '%s\\n' "\$DEFAULT_JVM_OPTS \$JAVA_OPTS \$${optsEnvironmentVar}" |
        xargs -n1 |
        sed ' s~[^-[:alnum:]+,./:=@_]~\\\\&~g; ' |
        tr '\\n' ' '
    )" '"\$@"'

```
https://github.com/gradle/gradle/blob/master/subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt

Reply:
This means we would need to add xargs, sed as dependency in base image. Do we want to do that? Or can we just go ahead with this change? This change will directly paste the args as is in java command

Reply:
```
/app # which xargs
/busybox/xargs
/app # which sed
/busybox/sed
/app #
```
Sorry, they are available, will incorporate this